File: java-sample-app/src/main/java/com/aminography/primeadapter/sample/tools/PackageUtils.java
package com.aminography.primeadapter.sample.tools;

import android.content.Context;
import android.content.Intent;

/**
 * Created by aminography on 8/10/2018.
 */
public class PackageUtils {

    public static void openApplication(Context context, String packageIdentifier) {
        Intent intent = context.getPackageManager().getLaunchIntentForPackage(packageIdentifier);
        if (intent != null) {
            intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
            context.startActivity(intent);
        }
    }

}

File: sample-app/src/main/java/com/aminography/primeadapter/sample/tools/PackageUtils.kt
package com.aminography.primeadapter.sample.tools

import android.content.Context
import android.content.Intent


/**
 * Created by aminography on 8/10/2018.
 */
class PackageUtils {

    companion object {

        fun openApplication(context: Context, packageIdentifier: String) {
            try {
                val intent = context.packageManager.getLaunchIntentForPackage(packageIdentifier)
                intent?.apply {
                    intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK
                    context.startActivity(this)
                }
            } catch (e: android.content.pm.PackageManager.NameNotFoundException) {
                e.printStackTrace()
            }
        }
    }

}
